Shepherd's review (#40)
core-wg/cf-reg-update: draft-ietf-core-cf-reg-update.md

  RFC7120: iana-early
  RFC7252: coap
  RFC9110: http-sema

>   RFC9193: senml-cf

Reviewer: I know nobody sees this, but s/cf/ct/

Author: done in [182822b](https://github.com/core-wg/cf-reg-update/pull/40/commits/182822b1aed08eb3453d1a76f911649a97b5b5c7)

This lack of guidance may engender confusion in both the registering party and the registrar, and has already led to erroneous registrations.

In {{iana}}, this document updates {{-coap}} by modifying the registration procedures for the "CoAP Content-Formats" registry to mitigate the risk of unintentional or malicious errors.
These updates amend the different portions of the registry, introduce a review procedure to be performed for most portions of the registry, and allow the registration of temporary Content-Format identifiers for certain portions of the registry.
This document also introduces a new column, "Media Type", to the registry.

# Conventions and Definitions

{::boilerplate bcp14-tagged-bcp14}

This document uses the terms "media type", "content coding", "content-type", and "content format" as defined in {{Section 2 of -senml-cf}}.

# Examples for Erroneous Registrations

This section contains examples of registration requests for a CoAP Content-Format with identifier 64999 in the FCFS portion of the "CoAP Content-Formats" registry, which must not be allowed to succeed.

> The following considerations also apply to alternative examples where, for the same combination of content type and content coding, a registration was requested for a CoAP Content-Format with identifier in the IETF Review or IESG Approval portion of the registry.

Reviewer: ```suggestion
The following considerations also apply for a registrations requested for a CoAP Content-Format with an identifier in the IETF Review or IESG Approval portion of the registry.
```

Reviewer 2: Sorry, what are the "considerations" here that the sentence talks about? Shouldn't it be "...the following examples..." ?

Or something like:

```
For each of the following example registration requests, a similar example can be constructed where the requested registration is for the CoAP Content-Format identifier 9999 in the IETF Review or IESG Approval portion of the registry.
```

Isn't that more clear and concrete (with number 9999) ?

Author: Trying to merge @cabo and @EskoDijk suggestions into a mad Frankenstein:

> _For each of the following example registration requests, one can create a similar instance where the requested registration is for a CoAP Content-Format identifier within the "IETF Review" or "IESG Approval" portion of the registry._

WDYT?

Author: done in [3d11d2b](https://github.com/core-wg/cf-reg-update/pull/40/commits/3d11d2b7bcd22c3a978a96225772eae8a1544067)

The 10000-64999 range now has two separate registration procedures.
If the registration consists solely of a registered media type name in the "Content Type" field, without any parameter names or "Content Coding", and the media type has not yet been used in this registry, then the policy is FCFS, as before.

> In all other cases, the policy is "Expert Review," following the procedure described in {{checks}}.

Reviewer 3: ```suggestion
In all other cases, the policy is "Expert Review", following the procedure described in {{checks}}.
```

A new column with the title "Note" has been added to the registry, which contains information about the expected review procedure.

## Temporary Content-Format Registrations

This section clarifies that the "CoAP Content-Formats" registry allows temporary registrations within the 0-255 and 256-9999 ranges.
The range 10000-64999 does not allow temporary registrations.

A temporary registration may be created for example by an IANA early allocation action, as requested by the authors of an Internet-Draft in the IETF stream.
Alternatively, it may be created because the referenced media type is still provisional (that is, included in the IANA "Provisional Standard Media Type" registry {{IANA.provisional-standard-media-types}}).

> A temporary registration is marked by an IANA note with the label "TEMPORARY" in the  "Content Type" column of the corresponding registry entry.

Reviewer 2: I think here naming the specific column is too detailed. IANA will mark it somewhere on the line, and that might change over time (e.g. do the label in the Reference column if they like that more).

Author: @EskoDijk what about:

> _A temporary registration is marked by IANA with the label "TEMPORARY" in the corresponding registry entry._

Author: done in [3d11d2b](https://github.com/core-wg/cf-reg-update/pull/40/commits/3d11d2b7bcd22c3a978a96225772eae8a1544067)
